frontend/institution/allInstitutionsController.js

```diff
@@ -12,9 +12,75 @@
 
         allInstitutionsCtrl.user = AuthService.getCurrentUser();
         allInstitutionsCtrl.isLoadingInstitutions = true;
+        allInstitutionsCtrl.allInstitutions = []
         allInstitutionsCtrl.institutions = [];
         allInstitutionsCtrl.filterKeyword = "";
 
+        allInstitutionsCtrl.allInstTab = true;
+        allInstitutionsCtrl.followingInstTab = false;
+        allInstitutionsCtrl.memberInstTab = false;
+        
+        const FOLLOWING_TAB = 'following';
+        const ALL_TAB = 'all';
+        
+        /**
+         * It sets the flags correctly according to the nextTab value.
+         * Besides, this function calls other aux functions that filter
+         * the institutions to achieve the expected behavior
+         * {string} nextTab
+         */
+        allInstitutionsCtrl.changeTab = function changeTab(nextTab) {
+            if (nextTab === ALL_TAB) {
+                allInstitutionsCtrl.allInstTab = true;
+                allInstitutionsCtrl.followingInstTab = false;
+                allInstitutionsCtrl.memberInstTab = false;
+                setAllInstitutions();
+            } else if (nextTab === FOLLOWING_TAB) {
+                allInstitutionsCtrl.allInstTab = false;
+                allInstitutionsCtrl.followingInstTab = true;
+                allInstitutionsCtrl.memberInstTab = false;
+                setFollowingInstitutions();
+            } else {
+                allInstitutionsCtrl.allInstTab = false;
+                allInstitutionsCtrl.followingInstTab = false;
+                allInstitutionsCtrl.memberInstTab = true;
+                setMemberInstitutions();
+            }
+        };
+
+        /**
+         * Sets the institutions to allInstitutions.
+         * It is done by cloning the allInstitutions array do avoid
+         * it to keep reference.
+         */
+        function setAllInstitutions() {
+            allInstitutionsCtrl.institutions = [...allInstitutionsCtrl.allInstitutions];
+        }
+
+        /**
+         * Sets institutions as an array with the institutions that the
+         * user follows.
+         */
+        function setFollowingInstitutions() {
+            allInstitutionsCtrl.institutions = allInstitutionsCtrl.allInstitutions.filter(inst => {
```

Comment:
I've realized that I cant set it to user.follows because it will be a problem to the pagination once user.follows has all the institutions the user is following... same thing to the member tab

Reply:
get pagination with filter from the backend. So you dont have to be filtering this  here.